### DISCUSSION TITLE
How can I PUT some files/directories into a github repository?

### DISCUSSION THREAD

**Discussion by elect86** [hub4j/github-api#1137] [Q&A] [answered]
As titled, I have some files which I'd like to upload into a github repository and I'd like to do that without having the project cloned locally.

I think I could manually dig myself into using `curl` as docs specifies [here](https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents)

However, I'd like to use some comfortable, such this library, if possible, but I havent figure it out yet how.

I guess `GHContent` may be relevant for my case, but I don't see any way I can set the `path` for example..

Can anyone help me?

Thanks in advance

**Comment by bitwiseman** [comment, marked as answer]
Here's an example based on [this test](https://github.com/hub4j/github-api/blob/d02f1946689b1b641254743752cccb6ce9229aa2/src/test/java/org/kohsuke/github/GHContentIntegrationTest.java#L142-L147) :

```
GHRepository ghRepository; // get your repo
GHContentBuilder ghContentBuilder = ghRepository.createContent()
    .message("Some commit msg"
    .path("some-path/to/where-you-want/yourFile.md")
    .content("yourContent")
    .commit();
```

**Reply by elect86** [reply]
Is there a possibility to upload binary files as well?

**Reply by bitwiseman** [reply]
I think this will do that: 

https://github.com/hub4j/github-api/blob/3f9954144a6df515bd7d6c8cdc123536430e893a/src/main/java/org/kohsuke/github/GHContentBuilder.java#L70-L73

Pass a `byte[]` to `content()`

**Reply by elect86** [reply]
Nice.

However what I'm really aiming to is uploading multiple files and commit once

Can Hub4j deal with blobs and trees, like [here](https://siddharthav.medium.com/push-multiple-files-under-a-single-commit-through-github-api-f1a5b0b283ae)?

**Comment by elect86** [comment]
Doesn't this

`GHRepository ghRepository; // get your repo`

Require the repository cloned?

**Reply by bitwiseman** [reply]
No. It is a REST call.  For example: 
   
```java
GHRepository ghRepository = gitHub.getRepository("hub4j/github-api");
```
